File: airportgap-typescript/tests/airports/listAirports.test.ts
import { describe, it, expect } from 'vitest'
import { ApiClient } from '../../src/http/apiClient'
describe('Airports list', () => {
  const api = new ApiClient()
  it('@smoke first page has 30 items and links', async () => {
    const r = await api.base.get('/airports')
    expect(r.status).toBe(200)
    const data = r.data?.data
    expect(Array.isArray(data)).toBe(true)
    expect(data.length).toBe(30)
    expect(r.data.links?.self).toBeTruthy()
    expect(r.data.links?.first).toBeTruthy()
    expect(r.data.links?.last).toBeTruthy()
  })
})

File: airportgap-java/src/test/java/spec/airports/ListAirportsTest.java
package spec.airports;

import config.Env;
import http.ApiClient;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import static io.restassured.RestAssured.given;
import static org.assertj.core.api.Assertions.assertThat;

public class ListAirportsTest {
  private final ApiClient api = new ApiClient(Env.baseUrl(), Env.token());

  @Test @Tag("smoke")
  void firstPageHasThirtyItemsAndLinks() {
    var resp = given()
        .spec(api.baseSpec())
        .get("/airports")
        .then()
        .statusCode(200)
        .extract().jsonPath();

    var data = resp.getList("data");
    assertThat(data).hasSize(30);
    assertThat(resp.getString("links.self")).isNotBlank();
    assertThat(resp.getString("links.first")).isNotBlank();
    assertThat(resp.getString("links.last")).isNotBlank();
  }
}
